### PR TITLE
Merge event's domain and name

### LINKED ISSUE #2994: Decide if event.domain separately from event.name is necessary
We mentioned the possibility of merging event.domain and event.name into one attribute. Now that even.domain is no longer on the Scope having it is a separate attribute is no longer mandatory.

There are a few things to consider here:
- Whether the combined attribute is good enough for use cases @tedsuo [mentioned](https://github.com/open-telemetry/opentelemetry-specification/issues/2897#issue-1416861012) ("primary index").
- If merged how do we define the formatting of the attribute value, e.g. should the value be prefixed by reverse FQDN? Can we also use plain prefixes like `k8s.`?
- What other operations need to be performed on the domain and name values which are qualitatively easier/possible when the values are recorded as separate attributes and are harder/impossible with a single combined attribute?

### PR DESCRIPTION
**Pull request by patrickhousley**
Fixes #2994

## Changes

Merging the `domain` and `name` fields for events and modifying language to refer to the first part of the name as `namespace`

### CODE REVIEW

File: specification/logs/event-api.md [hunk 1]
```diff
@@ -65,8 +66,6 @@ on `EventLogger`.
 
 * `logger` - the delegate [Logger](./bridge-api.md#logger) used to emit `Events`
   as `LogRecord`s.
-* `event_domain` - the domain of emitted events, used to set the `event.domain`
-  attribute.
```
Comment (jack-berg) [inline specification/logs/event-api.md:69]: We'll want to address the reference to event.domain in the [versioning and stability](https://github.com/open-telemetry/opentelemetry-specification/blob/92360f14caae693112944c088536f9e972600bbb/specification/versioning-and-stability.md#L214-L215) document as well.

File: specification/logs/event-api.md [hunk 2]
```diff
@@ -30,11 +30,12 @@ From OpenTelemetry's perspective LogRecords and Events are both represented
...
-Within a particular `domain`, the `name` uniquely defines a particular class or
-type of event. Events with the same `domain` / `name` follow the same schema
-which assists in analysis in observability platforms. Events are described in
-more detail in the [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md).
+LogRecords and Events: Events are LogRecords which have a unique `name` that
+includes a `namespace`. The `namespace` and name are separated by a period.
+The unique `name` defined a particular class or type of event. All events with
+the same `name` follow the same schema which assists in analysis in observability
+platforms. Events are described in more detail in the
+[semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md).
```
Comment (jack-berg) [inline specification/logs/event-api.md:38]: Let's keep details about the form of `name` in the semantic conventions event document. Including the details here requires that we maintain duplicate (or overlapping) copies of the same information which have to be kept in sync with each other. Notice how the metric api [name](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-name-syntax) documentation omits details about [metric naming guidelines from the semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md). We should follow the same pattern.
```suggestion
LogRecords and Events: Events are LogRecords which have a `name` which uniquely defines a particular class or type of event. All events with the same `name` follow the same schema which assists in analysis in observability platforms. Events are described in more detail in the [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md).
```

**Review by jack-berg — COMMENTED** [review]
Note that this change should be made in tandem with https://github.com/open-telemetry/semantic-conventions/pull/473.

File: specification/logs/event-api.md [hunk 3]
```diff
@@ -30,11 +30,7 @@ From OpenTelemetry's perspective LogRecords and Events are both represented
 using the same [data model](./data-model.md).
 
 However, OpenTelemetry does recognize a subtle semantic difference between
-LogRecords and Events: Events are LogRecords which have a `name` and `domain`.
-Within a particular `domain`, the `name` uniquely defines a particular class or
-type of event. Events with the same `domain` / `name` follow the same schema
-which assists in analysis in observability platforms. Events are described in
-more detail in the [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md).
+LogRecords and Events: Events are LogRecords which have a `name` which uniquely defines a particular class or type of event. All events with the same `name` follow the same schema which assists in analysis in observability platforms. Events are described in more detail in the [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md).
```
Comment (jack-berg) [inline specification/logs/event-api.md:33]: Can you use the [markdown style](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CONTRIBUTING.md#markdown-style) recommendation of limiting lines to 80 characters?

**Review by jack-berg — APPROVED** [review]
Can you add a [changelog](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md) entry?

Looks good to me otherwise.

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Now that https://github.com/open-telemetry/opentelemetry-specification/pull/3749 has been merged, is this ready to go?

File: CHANGELOG.md [hunk 4]
```diff
@@ -20,6 +20,10 @@ release.
 
 - Add definition for standard output log record exporter.
   ([#3741](https://github.com/open-telemetry/opentelemetry-specification/pull/3741))
+- BREAKING: Remove `event.domain` from log event attributes.
+  ([#3749](https://github.com/open-telemetry/opentelemetry-specification/pull/3749))
+- BREAKING: Change `event.name` definition to include `namespace`.
+  ([#3749](https://github.com/open-telemetry/opentelemetry-specification/pull/3749))
```
Comment (tigrannajaryan) [inline CHANGELOG.md:26]: Nit: I think this can be one item instead of two.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**